t/151-ssl-ctx.t

```diff
@@ -0,0 +1,384 @@
+# vim:set ft= ts=4 sw=4 et fdm=marker:
+
+use Test::Nginx::Socket::Lua;
+use Cwd qw(cwd);
+use Digest::MD5 qw(md5_hex);
+
+repeat_each(3);
+
+plan tests => repeat_each() * (blocks() + 2);
+
+our $CWD = cwd();
+$ENV{TEST_NGINX_LUA_PACKAGE_PATH} = "$::CWD/lib/?.lua;;";
+$ENV{TEST_NGINX_HTML_DIR} ||= html_dir();
+our $TEST_NGINX_LUA_PACKAGE_PATH = $ENV{TEST_NGINX_LUA_PACKAGE_PATH};
+our $TEST_NGINX_HTML_DIR = $ENV{TEST_NGINX_HTML_DIR};
+
+log_level 'debug';
+
+no_long_string();
+
+sub read_file {
+    my $infile = shift;
+    open my $in, $infile
+        or die "cannot open $infile for reading: $!";
+    my $cert = do { local $/; <$in> };
+    close $in;
+    $cert;
+}
+
+our $clientKey = read_file("t/cert/ca-client-server/client.key");
+our $clientUnsecureKey = read_file("t/cert/ca-client-server/client.unsecure.key");
+our $clientCrt = read_file("t/cert/ca-client-server/client.crt");
+our $clientCrtMd5 = md5_hex($clientCrt);
+our $serverKey = read_file("t/cert/ca-client-server/server.key");
+our $serverUnsecureKey = read_file("t/cert/ca-client-server/server.unsecure.key");
+our $serverCrt = read_file("t/cert/ca-client-server/server.crt");
+our $caKey = read_file("t/cert/ca-client-server/ca.key");
+our $caCrt = read_file("t/cert/ca-client-server/ca.crt");
+our $http_config = <<_EOS_;
+lua_package_path "\$prefix/html/?.lua;$TEST_NGINX_LUA_PACKAGE_PATH/?.lua;;../lua-resty-lrucache/lib/?.lua;";
+
+init_by_lua_block {
+    local ffi = require "ffi"
```

Comment:
Yup, these tests actually adds maintenance overhead for me, Since these tests is the same as lua-resty-core. According what you have said, Can I think that if we are adding the FFI C API, we should simply test it in `lua-nginx-module`  and specially test in `lua-resty-core` ?